packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -28,20 +33,38 @@ export interface CSSOutput {
  * @param state Babel state - should house options and meta data used during the transformation.
  */
 const getInterpolation = <TNode extends {}>(expression: TNode | undefined, state: State) => {
-  if (t.isIdentifier(expression) && state.declarations) {
-    const declaration = state.declarations[expression.name];
-    if (t.isVariableDeclaration(declaration) && declaration.kind === 'const') {
-      const potentialValue = declaration.declarations[0].init;
-      if (
-        t.isStringLiteral(potentialValue) ||
-        t.isNumericLiteral(potentialValue) ||
-        t.isObjectExpression(potentialValue)
-      ) {
-        return potentialValue;
-      }
+  if (!state.declarations) {
+    return expression;
+  }
+
+  let value: t.Node | undefined | null = undefined;
+
+  if (t.isIdentifier(expression)) {
+    const binding = state.declarations[expression.name];
+
+    if (t.isVariableDeclaration(binding) && binding.kind === 'const') {
+      value = binding.declarations[0].init;
+    }
+  }
+
+  if (t.isMemberExpression(expression)) {
+    const bindingName = getMemberExpressionIdentifier(expression);
```

Comment:
Instead of `getMemberExpressionIdentifier `, `getMemberExpressionPath `, `getValueFromObjectExpression` can't we do something like this:

```typescript
let bindingName;
let bindingKey;

traverse(expression, {
  MemberExpression: {
    enter(path) {
      if (t.isIdentifier(path.node.object)) {
       // Will be theme
        bindingName = path.node.object.name;
      }
    },
    exit(path) {
      if (t.isIdentifier(path.node.property)) {
       // Will be colors, light, primary. But as we are mutating, we will take last value, i.e. primary
        bindingKey = path.node.property.name;
      }
    }
  }
});

// and then probably

const binding = state.declarations[bindingName];
let value;

traverse(binding.declarations[0].init, {
  ObjectProperty(path) {
      if (t.isIdentifier(path.node.key, { name: bindingKey })) {
         value = path.node.value;
      }
    }
});
```

Not sure, I can be wrong here 😞 (I am new to babel plugin world). Just thought of leveraging the power of babel traverse instead of our own tree traversal. 🤔 This might save some code and will be faster than `getValueFromObjectExpression` where we have loop inside another loop. 🤔

Reply:
ohh not a bad idea - let me play with it. the only downside i see atm is:

1. it won't traverse the top level node (any ideas how we can without wrapping it in something?)
2. we need access to the parent path - so need to update a lot of wrapping code (which we will definitely do in the later PR for accessing the binding instead of the declaration stuff)

Reply:
oh right, `traverse` needs a parent node.

🤔 I think this should work

```typescript
traverse(binding.declarations[0].init, {
  ObjectProperty(path) {
      if (t.isIdentifier(path.node.key, { name: bindingKey })) {
         value = path.node.value;
      }
    }
});
```

because `binding.declarations[0].init` will be `ObjectExpression` and we are looking for child node i.e. `ObjectProperty`.

For other one, we can skip exit path and pick `primary` color from `MemberExpression` `property` key and `theme` will come when we traverse children or may be checking `object` key when we have case like `theme.primary`. Something like this:
```typescript
const bindingKey = t.isMemberExpression(expression) && t.isIdentifier(expression.property) && expression.property.name;
let bindingName = t.isMemberExpression(expression) && t.isIdentifier(expression.object) && expression.object.name;

traverse(expression, {
 //  Will not run in case of `theme.primary`. But will run when we have `theme.colors.light` and so on
  MemberExpression(path) {
    if (t.isIdentifier(path.node.object)) {
      // Will be theme.
      bindingName = path.node.object.name;
    }
  }
});
```

Reply:
And regarding parent path, I found this:

```typescript
traverse(expression, {
  noScope: true, // <====== this is the magic key, it says there is no scope, so no parent path is required
  MemberExpression(path) {
    if (t.isIdentifier(path.node.object)) {
      bindingName = path.node.object.name;
    }
  }
});
```

Reply:
ohh!! cool will have a play!! good job

Reply:
sounds like we will only need parent path (and scope) for when we want to run evaluate.

Reply:
Seems like. Because it might need to find the binding (if it's a variable) in that scope. But rest of the traversal can be done without scope and parent path right.

Reply:
okay so i've made some changes - one current problem is we need more information that just a single identifier to get the property value - because you can have the same identifier name in different parts of the object expression. i'll have a dig and see if theres anything in the scope/path we can use.